### PR TITLE
chore(docs): Add CockroachDB deployment guide

### PR DESCRIPTION
**Pull request by nareshdesh**
Adding CockroachDB deployment with openebs-localpv storage provisioner

Signed-off-by: Naresh Deshaveni <naresh.deshaveni@mayadata.io>

### CODE REVIEW

File: docs/cockroachdb.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+  ```bash
+  root@:26257/defaultdb> CREATE DATABASE sbtest;
+  root@:26257/defaultdb> \q
+  ```
+ 
+#### Accessing the database's web interface. 
+We will be using NodePort for accessing the service.
+In production environment either use loadbalancer or ingress services as per your requirement
+
+- Create a new node port service using the following
```
Comment (ranjithwingrider) [inline docs/cockroachdb.md:242]: Create a new node port service using the following > Create a new node port service using the following YAML spec. The following sample spec is saved as `cockroachdb-public-node-port.yaml`.

File: docs/cockroachdb.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+  ```bash 
+  cat cockroachdb-public-node-port.yaml
```
Comment (ranjithwingrider) [inline docs/cockroachdb.md:244]: remove 'cat cockroachdb-public-node-port.yaml'

File: docs/cockroachdb.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+
+## Configuration workflow 
+
+1. [Install OpenEBS]((/docs/next/cockroachdb.html#instal-openebs)
+2. [Select OpenEBS storage engine](/docs/next/cockroachdb.html#select-openebs-storage-engine) 
+3. [Configure OpenEBS Local PV StorageClass](/docs/next/cockroachdb.html#configure-openebs-local-pv-storageclass) 
+4. [Install CockroachDB Operator](/docs/next/cockroachdb.html#install-cockroachdb-operator)
+5. [Accessing CockroachDB](/docs/next/cockroachdb.html#accessing-cockroachdb)
+6. [Testing CockroachDB Performance on OpenEBS](/docs/next/cockroach.html#testing-cockroachdb-performance-on-openebs)
```
Comment (ranjithwingrider) [inline docs/cockroachdb.md:39]: Let's remove item number 6. It's not required.

File: docs/cockroachdb.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+  ```bash
+  https://<any_node_external-ip>:<NodePort>
+  ```
+  Login credentials for the web UI
+  Username: `roach`
+  Password: `Q7gc8rEdS`
+
+
+      
+### Testing CockroachDB Performance on OpenEBS
```
Comment (ranjithwingrider) [inline docs/cockroachdb.md:293]: Let's remove the performance tests details. We can keep it under the solution guide only.

**Review by ranjithwingrider — COMMENTED** [review]
@nareshdesh Please find my comments.

File: docs/cockroachdb.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+sidebar_label: CockroachDB
+---
+------
+
+<img src="/docs/assets/o-cockroachdb.png" alt="OpenEBS and CockroachDB" style="width:400px;">
+
+
+## Introduction
+
+CockroachDB is a cloud-native SQL database for building global, scalable cloud services that survive disasters. It is a distributed SQL database built on a transactional and strongly-consistent key-value store. It scales horizontally; survives disk, machine, rack, and even datacenter failures with minimal latency disruption and no manual intervention; supports strongly-consistent ACID transactions; and provides a familiar SQL API for structuring, manipulating, and querying data.[1](https://github.com/cockroachdb/cockroach) [2](https://www.cockroachlabs.com/docs/stable/orchestrate-cockroachdb-with-kubernetes.html).
```
[the remark this answers is not in the document]
Comment (nareshdesh) [reply docs/cockroachdb.md:13]: Part of the introduction content is taken from the sites, hence have kept the reference link

[the remark this answers is not in the document]
Comment (ranjithwingrider) [reply docs/cockroachdb.md:13]: Yes I know, We don't need a reference link in Documentation.

File: docs/cockroachdb.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+<img src="/docs/assets/svg/Local-PV-Distributed-device-cockroachdb.svg" alt="OpenEBS and CockroachDB" style="width:100%;">
+
+<br>
+
+The Local PV volume will be provisioned on a node where CockroachDB pods are getting scheduled and uses one of the matching unclaimed block device for each of them, which will then use the entire block device for storing data. No other application can use this device. If users have limited blockdevices attached to some nodes, they can use `nodeSelector` in the application YAML to provision applications on particular nodes where the available block device is present. The recommended configuration is to have at least three nodes and one unclaimed external disk to be attached per node. 
+
+
+## Configuration workflow 
+
+1. [Install OpenEBS](/docs/next/cockroachdb.html#instal-openebs)
```
Comment (ranjithwingrider) [inline docs/cockroachdb.md:34]: Hyperlink is still not correct.. Need to update the following as the correct link. 
/docs/next/cockroachdb.html#install-openebs

**Review by ranjithwingrider — CHANGES_REQUESTED** [review]
@nareshdesh Sorry. I missed one more error in the hyperlink of step 1 under configuration workflow. The spelling of install was not correct in the link. The rest of the contents are looks good to me.

File: docs/cockroachdb.md (hunk 6 above)
Comment (nareshdesh) [reply docs/cockroachdb.md:34]: thank you @ranjithwingrider for finding it out.

**Review by ranjithwingrider — APPROVED** [review]

**Pull request merged**